packages/bsky/src/api/app/bsky/graph/getSuggestedFollowsByActor.ts

```diff
@@ -0,0 +1,135 @@
+import { sql } from 'kysely'
+import { Server } from '../../../../lexicon'
+import AppContext from '../../../../context'
+import { InvalidRequestError } from '@atproto/xrpc-server'
+
+const MAX_RESULTS_LENGTH = 10
+
+export default function (server: Server, ctx: AppContext) {
+  server.app.bsky.graph.getSuggestedFollowsByActor({
+    auth: ctx.authVerifier,
+    handler: async ({ auth, params }) => {
+      const { actor } = params
+      const viewer = auth.credentials.did
+
+      const db = ctx.db.getReplica()
+      const actorService = ctx.services.actor(db)
+      const actorDid = await actorService.getActorDid(actor)
+
+      if (!actorDid) {
+        throw new InvalidRequestError('Actor not found')
+      }
+
+      const actorsViewerFollows = db.db
+        .selectFrom('follow')
+        .where('creator', '=', viewer)
+        .select('subjectDid')
+
+      /**
+       * 20 most liked accounts that aren't already followed by the viewer, ARE
+       * the viewer, or are the actor
+       */
+      const mostLikedAccounts = db.db
+        .selectFrom(
+          db.db
+            .selectFrom('like')
+            .where('creator', '=', actorDid)
+            .select(sql`split_part(subject, '/', 3)`.as('subjectDid'))
+            .as('likes'),
+        )
+        .select('likes.subjectDid as did')
+        .select((qb) => qb.fn.count('likes.subjectDid').as('count'))
+        .where('likes.subjectDid', 'not in', actorsViewerFollows)
+        .where('likes.subjectDid', 'not in', [actorDid, viewer])
+        .groupBy('likes.subjectDid')
+        .orderBy('count', 'desc')
+        .limit(20)
+
+      /**
+       * 20 most liked accounts as actor results
```

Comment:
is there a reason we get 20 when the max is 10?

Reply:
Oh just to allow for some of the results to be blocked/muted by the viewer. I'm assuming that the # of users that block > 50% of the result set is pretty rare.

Basically I want it to be able to fill 10 results as much as possible. We could filter by blocks in each of the queries too, though 🤔

Reply:
Answered my own question here, one sec